packages/TelegramClient-Core.package/TCCChat.class/class/newFromChatEvent..st

```diff
@@ -1,12 +1,13 @@
 instance creation
 newFromChatEvent: anEvent
 
-	| chat |
-	chat := anEvent at: 'chat'.
-	^ self new
-		title: (chat at: 'title');
-		id: (chat at: 'id');
-		canSendMessages: ((chat at: 'permissions') at: 'can_send_messages');
-		type: ((chat at: 'type') at: '@type');
-		lastMessage: '';
+	| type chat |
+	type := ((anEvent at: 'chat') at: 'type') at: '@type'.
+	
+	((type = 'chatTypeSupergroup') or: (type = 'chatTypeBasicGroup'))
+		ifTrue: [chat := TCCGroup new]
```

Comment:
a bit weird to create a chat and then initialize it, why not use a constructor here? 
If true, return a TCCGroup from the block, if false return a `TCCChat`. Construct both in their respective blocks

Reply:
Do you mean something like this?
```
newFromChatEvent: anEvent

	| type |
	type := ((anEvent at: 'chat') at: 'type') at: '@type'.
	
	((type = 'chatTypeSupergroup') or: (type = 'chatTypeBasicGroup'))
		ifTrue: [^ TCCGroup new initializeFromChatEvent: anEvent.]
		ifFalse: [^ self new initializeFromChatEvent: anEvent.].
```
We could override `TCCGroup >> newFromChatEvent:` to use that directly but we would have to write a second constructor for the `TCCChat` to do that there too, which seemed a bit weird because we would then have the `newFromChatEvent:` just for the decision whether its a group or not.
What do you suggest would be the best way?